addons/mass_mailing_hr/static/src/fields/html_field/mass_mailing_html_field.js

```diff
@@ -0,0 +1,16 @@
+import { patch } from "@web/core/utils/patch";
+import { MassMailingHtmlField } from "@mass_mailing/fields/html_field/mass_mailing_html_field";
+import { useRecordObserver } from "@web/model/relational_model/utils";
+
+patch(MassMailingHtmlField.prototype, {
+    setup() {
+        super.setup();
+        useRecordObserver(async (record) => {
+            // consult the mailing_model_name field so it can be recognized by the observer
+            const _ = record.data.mailing_model_name;
+            if (this.editor && !this.editor.isDestroyed && this.editor.isReady) {
+                this.editor.dispatchTo("on_mailing_model_updated_handlers");
+            }
+        });
+    },
+});
```

Comment:
Why a patch and not directly modify MassMailingHtmlField? It seems legit to me to add directly a dispatchTo inside the recordObserver.